modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -186,6 +186,348 @@ Test_ProcessTimes(void)
+                                       sizeof(QUOTA_LIMITS),
+                                       NULL);
+    ok_hex(Status, STATUS_DATATYPE_MISALIGNMENT);
+
+    /* Buffer too small */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS) - 1,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Right buffer size but NULL pointer */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS),
+                                       NULL);
+    ok_hex(Status, STATUS_ACCESS_VIOLATION);
+
+    /* Buffer too large */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS) + 1,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Buffer too small, ask for length */
+    Length = 0x55555555;
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       NULL,
+                                       sizeof(QUOTA_LIMITS) - 1,
+                                       &Length);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+    ok_dec(Length, 0x55555555);
+
+    /* Valid parameters, no return length */
+    RtlFillMemory(&QuotaLimits1, sizeof(QuotaLimits1), 0x55);
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       &QuotaLimits1,
+                                       sizeof(QUOTA_LIMITS),
+                                       NULL);
+    ok_hex(Status, STATUS_SUCCESS);
+
+    /* Trace the returned data (1) */
+    trace("QuotaLimits1.PagedPoolLimit = %Iu\n", QuotaLimits1.PagedPoolLimit);
+    trace("QuotaLimits1.NonPagedPoolLimit = %Iu\n", QuotaLimits1.NonPagedPoolLimit);
+    trace("QuotaLimits1.MinimumWorkingSetSize = %Iu\n", QuotaLimits1.MinimumWorkingSetSize);
+    trace("QuotaLimits1.MaximumWorkingSetSize = %Iu\n", QuotaLimits1.MaximumWorkingSetSize);
+    trace("QuotaLimits1.PagefileLimit = %Iu\n", QuotaLimits1.PagefileLimit);
+    trace("QuotaLimits1.TimeLimit = %I64u\n", QuotaLimits1.TimeLimit.QuadPart);
+
+    /* Again, this time with a return length */
+    Length = 0x55555555;
+    RtlFillMemory(&QuotaLimits2, sizeof(QuotaLimits2), 0x55);
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessQuotaLimits,
+                                       &QuotaLimits2,
```

Comment:
Why using a separate variable, instead of re-using the QuotaLimits1 one (after resetting its contents)?

Reply:
ok thanks; when comparisons between different quota calls will be useful, these two variables could be reintroduced.